Rewrite Videos from YouTube (#3851)
ppy/osu-wiki: wiki/Guides/Videos_from_YouTube/en.md

> ---
> tags:
>   - how to download videos from YouTube
>   - how to use YouTube videos
> ---

Reviewer: ```suggestion
---
tags:
  - how to download videos from YouTube
  - how to use YouTube videos
  - how to
  - guide
  - tutorial
---
```

# Videos from YouTube

> [YouTube](https://youtube.com) is a very convenient resource. If a song has a music video, it will most likely be on YouTube, and probably compressed at that golden point where the video looks decent but will not be a pain to download.

Reviewer: ```suggestion
[YouTube](https://youtube.com "YouTube") is a very useful resource for creators looking to add background videos to their [beatmaps](/wiki/Beatmaps): if a song has a music video, it will most likely be on YouTube. (And it will probably be compressed at that golden point where the video looks decent but will not be a pain to download.)

There are many ways to obtain videos from YouTube onto yoru device. Below, we've listed a few of the most popular ways of obtaining such videos:
```
Maybe try this?

> This tutorial is intended for those who **can not** find the video elsewhere but YouTube.

Reviewer: not sure if this line is needed

Author: I think we want less videos sourced from yt and we have this guide as a last resort, but I'm not sure.

## Using online tools

> [ytmp3.cc](https://ytmp3.cc) is an online YouTube converter service. Using it is very straightforward: paste the YouTube video link into the text box, select `mp4`, and click `Convert`. After conversion is finished, click `Download` to obtain the video file.

Reviewer: ```suggestion
[YouTube to Mp3](https://ytmp3.cc "YouTube to Mp3") is an online video converter service designed for YouTube. Using it is very straightforward: first, paste your desired YouTube video link into the text box; then, select `mp4`, click `Convert`; and after the conversion is finished, click `Download` to obtain the video file.
```
*Do we need to add any legal warnings abt this?*

> [ytmp3.cc](https://ytmp3.cc) is an online YouTube converter service. Using ytmp3.cc is very straightforward. Paste the YouTube video link into the text box, select "mp4", and click "Convert". After conversion is finished, click "Download" to download the video file.

Reviewer 2: ```suggestion
[ytmp3.cc](https://ytmp3.cc) is an online YouTube converter service. Using it is very straightforward: paste the YouTube video link into the text box, select `mp4`, and click `Convert`. After conversion is finished, click `Download` to obtain the video file.
```

## Using youtube-dl

> [youtube-dl](https://youtube-dl.org) is a command-line program to download videos from YouTube and other sites. The fastest way to download from YouTube using youtube-dl is by using the `-f bestvideo` option:

Reviewer: ```suggestion
[youtube-dl](https://youtube-dl.org "youtube-dl") is a [command-line](https://en.wikipedia.org/wiki/Command-line_interface "Wikipedia") program to download videos from YouTube and other sites. The fastest way to download from YouTube using youtube-dl is by using the `-f bestvideo` option:
```

```cmd

> youtube-dl.exe -f bestvideo <video link>

Reviewer 2: I think that for these who badly want to use it, but don't quite know where to past the command, you can mention the terminal, or the console, or PowerShell.

Reviewer: Am willing to provide screenshots for this if needed

```

## Removing the audio tracks

> *Main page: [Compressing Files](/wiki/Guides/Compressing_Files)*

Reviewer: ```suggestion
*Fora full guide on how to remove audio trakcs from videos, see: [Compressing Files](/wiki/Guides/Compressing_Files)*
```

> After downloading the video, you can load the video into the beatmap as you would normally through the [design tab](wiki/Beatmap_editor/Design) in the [beatmap editor](wiki/Beatmap_editor). However, the video will contain audio tracks that will be unused and will take up extra space. See [Compressing Files](/wiki/Guides/Compressing_Files) for ways to remove audio from the video.

Reviewer 2: it's **also** worth mentioning instead that videos with sound not only take space, but are also forbidden by the ranking criteria -- for an end user as a mapper this is more important and down to earth
```suggestion
After downloading the video, you can load the video into the beatmap as you would normally through the [design tab](wiki/Beatmap_editor/Design) in the [beatmap editor](wiki/Beatmap_editor). However, the video will contain an audio track that takes up extra space. See [Compressing Files](/wiki/Guides/Compressing_Files) for ways to remove audio from the video.
```

> After downloading the video, you can load the video into the beatmap as you would normally through the [design tab](wiki/Beatmap_editor/Design) in the [beatmap editor](wiki/Beatmap_editor). However, not only will the video contain audio tracks that will be unused and take up extra space, the beatmap [Ranking Criteria](/wiki/Ranking_Criteria#Video-and-background) forbids videos with audio tracks as they are not used. See [Compressing Files](/wiki/Guides/Compressing_Files) for ways to remove audio from the video.

Reviewer: ```suggestion
After downloading the video, you can load the video into the beatmap as you would normally through the [design tab](wiki/Beatmap_editor/Design) in the [beatmap editor](wiki/Beatmap_editor). However, not only will the video contain audio tracks that will be unused and take up extra space, the beatmap [Ranking Criteria](/wiki/Ranking_Criteria#Video-and-background) forbids videos with audio tracks, as they are not actually used in-game.
```
